libcxx/include/variant

```diff
@@ -340,21 +345,20 @@ struct _LIBCPP_TEMPLATE_VIS variant_alternative<_Ip, variant<_Types...>> {
 
 inline constexpr size_t variant_npos = static_cast<size_t>(-1);
 
-_LIBCPP_HIDE_FROM_ABI constexpr int __choose_index_type(unsigned int __num_elem) {
-  if (__num_elem < numeric_limits<unsigned char>::max())
-    return 0;
-  if (__num_elem < numeric_limits<unsigned short>::max())
-    return 1;
-  return 2;
+template <size_t _NumAlternatives>
```

Comment:
Consider returning `integral_constant<T, 0>` so that we don't have to worry about type conversions, promotions, etc.

Reply:
Since this is only an implementation detail for `__variant_index_t` this seems a bit overkill to me.